include/pops/simulation.hpp

```diff
@@ -521,6 +637,9 @@ class Simulation
                 else {
                     dispersers_from_cell = lambda * infected(i, j);
                 }
+                if (soil_pool_ && soil_pool_->active())
+                    dispersers_from_cell +=
+                        soil_pool_->dispersers_from(i, j, generator_);
```

Comment:
Can only move one cell, so generate separately from the other disperses and infect separately in one cell.

Reply:
Disperses from soil are now retrieved in disperse (so not generated in the current terminology, but obtained from the storage).